source/adapters/hip/enqueue.cpp

```diff
@@ -1516,10 +1500,18 @@ urEnqueueUSMAdvise(ur_queue_handle_t hQueue, const void *pMem, size_t size,
       // the runtime.
       if (Result == UR_RESULT_ERROR_INVALID_ENUMERATION) {
         releaseEvent();
-        setErrorMessage("mem_advise is ignored as the advice argument is not "
-                        "supported by this device",
-                        UR_RESULT_SUCCESS);
-        return UR_RESULT_ERROR_ADAPTER_SPECIFIC;
+        // UR_RESULT_ERROR_INVALID_ENUMERATION is returned when using a valid
+        // but currently unmapped advice arguments as not supported by this
+        // platform. Therefore, warn the user instead of throwing and aborting
+        // the runtime.
+        if (Result == UR_RESULT_ERROR_INVALID_ENUMERATION) {
```

Comment:
I may be missing something here but is this `if (Result == UR_RESULT_ERROR_INVALID_ENUMERATION)` meant to be checked again inside the above `if (Result == UR_RESULT_ERROR_INVALID_ENUMERATION)`? It is already true in this case. Is it meant to be wrapped in `if (Result != UR_RESULT_SUCCESS)` -> check if invalid else `throw`?

Reply:
I think I messed up the rebase here, now it just falls through to `UR_CHECK_ERROR(Result)` if we aren't doing the adapter specific thing

Reply:
Yeah that makes sense to me now. Thanks, I actually thuoght it must've been a dodgy rebase by the looks of it.